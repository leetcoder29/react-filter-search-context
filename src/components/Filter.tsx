import React, { FC, useContext, ChangeEvent } from 'react';
import { SearchContext } from '../App';

export const Filter: FC = () => {
  const context = useContext(SearchContext);

  const handleSelectChange = (e: ChangeEvent<HTMLSelectElement>) => {
    context!.handleSetFilter(e.target.value);
  };

  return (
    <div className='filter-container'>
      <label htmlFor='filter'>
        Filter
        <select onChange={handleSelectChange}>
          <option value='FILTER_OFF'>All</option>
          <option value='FILTER_ERRAND_ONLY'>Errand</option>
          <option value='FILTER_WORK_ONLY'>Work</option>
        </select>
      </label>
    </div>
  );
};
