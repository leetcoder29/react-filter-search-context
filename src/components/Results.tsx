import React, { useContext } from 'react';
import { SearchContext, Todo } from '../App';

export const Results: React.FC = () => {
  const context = useContext(SearchContext);

  const renderResults = () => {
    return context!.searchState.searchResults
      .filter((item: Todo) => {
        const searchFilter = context!.searchState.searchFilter;
        const { category } = item;
        if (searchFilter === 'FILTER_OFF') return true;
        if (searchFilter === 'FILTER_WORK_ONLY' && category === 'work')
          return true;
        if (searchFilter === 'FILTER_ERRAND_ONLY' && category === 'errand')
          return true;
      })
      .map((item: Todo) => {
        const { title, body, id } = item;
        return (
          <div key={id} className='item-container'>
            <h3 className='item-title'>{title}</h3>
            <p className='item-body'>{body}</p>
          </div>
        );
      });
  };

  return (
    <div>
      <h2>Results</h2>
      {context!.searchState.searchTerm ? renderResults() : null}
    </div>
  );
};
