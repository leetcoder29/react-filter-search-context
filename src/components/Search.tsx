import React, { FC, ChangeEvent, useState, useContext } from 'react';
import { SearchContext } from '../App';

export const Search: FC = () => {
  const [searchTerm, setSearchTerm] = useState('');
  const context = useContext(SearchContext);

  const handleSetSearchTerm = (e: ChangeEvent<HTMLInputElement>) => {
    const value = e.target.value;
    setSearchTerm(value);
  };

  const handleFormSubmit = (e: ChangeEvent<HTMLFormElement>) => {
    e.preventDefault();
    context!.handleSearchSubmit(searchTerm);
  };

  return (
    <div>
      <form onSubmit={handleFormSubmit}>
        <label htmlFor='search-term'>
          Search <input onChange={handleSetSearchTerm} id='username' />
        </label>
        <button type='submit'>Search</button>
      </form>
    </div>
  );
};
