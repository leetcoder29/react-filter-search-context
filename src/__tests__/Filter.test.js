import React from 'react';
import { Filter } from '../components/Filter';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJSON from 'enzyme-to-json';
configure({ adapter: new Adapter() });

describe('<Filter />', () => {
  it('renders and displays labels with text properly', () => {
    const wrapper = shallow(<Filter />);
    const label = wrapper.find('label');
    expect(label.text()).toBe('FilterAllErrandWork');
    const select = wrapper.find('select');
    expect(select.children()).toHaveLength(3);
  });

  it('renders and matches the snapshot', () => {
    const wrapper = shallow(<Filter />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
