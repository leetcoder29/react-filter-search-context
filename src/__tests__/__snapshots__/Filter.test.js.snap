// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Filter /> renders and matches the snapshot 1`] = `
<div
  className="filter-container"
>
  <label
    htmlFor="filter"
  >
    Filter
    <select
      onChange={[Function]}
    >
      <option
        value="FILTER_OFF"
      >
        All
      </option>
      <option
        value="FILTER_ERRAND_ONLY"
      >
        Errand
      </option>
      <option
        value="FILTER_WORK_ONLY"
      >
        Work
      </option>
    </select>
  </label>
</div>
`;
